src/python/pants/backend/python/dependency_inference/default_module_mapping.py

```diff
@@ -67,13 +93,17 @@ def two_groups_hyphens_two_replacements_with_suffix(
         for custom_suffix in ("", "_v1", "_v2", "_v3")
     ],
     re.compile(r"""^(opentelemetry-instrumentation-)([^.]+)"""): [
-        partial(two_groups_hyphens_two_replacements_with_suffix, second_group_replacement="_"),
+        partial(
+            two_groups_hyphens_two_replacements_with_suffix,
+            second_group_replacement=PackageSeparator.UNDERSCORE,
+        ),
     ],
-    re.compile(r"""^(oslo-.+)"""): [first_group_hyphen_to_underscore],
+    re.compile(r"""^oslo-.+"""): [all_hyphen_to_underscore],
```

Comment:
this is identical in logic; and also tested [here](https://github.com/pantsbuild/pants/pull/20513/files#diff-0dfccf1413aeedaf2cab49c919b8f8d79d7e995cf0715af3de915784b19e5d3fR977)